SOURCES/etc/xapi.d/plugins/smartctl.py

```diff
@@ -10,31 +10,30 @@
 from xcpngutils.operationlocker import OperationLocker
 
 @error_wrapped
-def _list_disks():
-    disks = []
+def _list_devices():
+    devices = []
     result = run_command(['smartctl', '--scan'])
     for line in result['stdout'].splitlines():
-        if line.startswith('/dev/') and not line.startswith('/dev/bus/'):
-            disks.append(line.split()[0])
-    return disks
+        devices.append({'name': line.split()[0], 'type':line.split()[2]})
+    return devices
 
 @error_wrapped
 def get_information(session, args):
     results = {}
     with OperationLocker():
-        disks = _list_disks()
-        for disk in disks:
-            cmd = run_command(["smartctl", "-j", "-a", disk], check=False)
-            results[disk] = json.loads(cmd['stdout'])
+        devices = _list_devices()
+        for device in devices:
+            cmd = run_command(["smartctl", "-j", "-a", "-d", device['name'], device['type']], check=False)
+            results[device] = json.loads(cmd['stdout'])
```

Comment:
This shouldn't work because device is a dict and cannot be used as a key. The problem with megraid is that we now have 2 devices with the same name. Something like:
```
/dev/bus/0 -d megaraid,0 # /dev/bus/0 [megaraid_disk_00], SCSI device
/dev/bus/0 -d megaraid,1 # /dev/bus/0 [megaraid_disk_01], SCSI device
```
So if we use the name like it is done in the current version we will have the second `/dev/bus/0` that will overwrite the first one. So in XO we will see information and health but at least info will only be the last device... But maybe it is ok. Otherwise I think about using a key like `/dev/bus/0@megaraid,1`. I'm currently checking with XO team what can be done. For now just use `device["name"]` I think.

Reply:
With my old program that works with increments of i, it worked.

Reply:
I see that from the xe host-call.
![image](https://github.com/user-attachments/assets/70ff6d97-bf6e-4a2c-b0b8-123697de6c4b)